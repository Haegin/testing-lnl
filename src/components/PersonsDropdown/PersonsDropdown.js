import React from 'react'

const PersonsDropdown = ({ persons, ...rest }) => (
  <select name="person" {...rest}>
    {persons.map(person => (
      <option value={person.name} key={person.name}>
        {person.name}
      </option>
    ))}
  </select>
)

export default PersonsDropdown
