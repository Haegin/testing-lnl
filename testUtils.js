import React from 'react'
import { render as rendar } from 'react-testing-library'
import { BrowserRouter } from 'react-router-dom'

const render = component => rendar(<BrowserRouter>{component}</BrowserRouter>)

export { render }
export * from 'react-testing-library'
