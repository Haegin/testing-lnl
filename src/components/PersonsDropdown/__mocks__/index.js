import React from 'react'

import PersonsDropdown from '../PersonsDropdown'

const DATA = {
  persons: [
    { name: 'Luke Skywalker' },
    { name: 'R2D2' },
    { name: 'C-3PO' },
    { name: 'Darth Vader' },
    { name: 'Yoda' },
  ],
}

const MockPersonsDropdown = props => <PersonsDropdown {...props} {...DATA} />

export default MockPersonsDropdown
