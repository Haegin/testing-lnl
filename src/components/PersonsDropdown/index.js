import { compose, branch, renderComponent, mapProps } from 'recompose'
import { graphql } from 'react-apollo'

import personsQuery from '../../queries/personsQuery'
import Loading from '../Loading'

import PersonsDropdown from './PersonsDropdown'

export default compose(
  graphql(personsQuery),
  branch(({ data: { loading } }) => loading, renderComponent(Loading)),
  mapProps(({ data, ...rest }) => ({ persons: data.allPersons, ...rest }))
)(PersonsDropdown)
