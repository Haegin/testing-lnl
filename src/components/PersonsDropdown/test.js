import React from 'react'
import { render } from '../../../testUtils'

import PersonsDropdown from './'

jest.mock('./')

describe('PersonsDropdown', () => {
  test('it matches the snapshot', () => {
    const { container } = render(<PersonsDropdown />)
    expect(container).toMatchSnapshot()
  })
})
