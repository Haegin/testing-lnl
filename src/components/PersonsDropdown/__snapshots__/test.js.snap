// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PersonsDropdown it matches the snapshot 1`] = `
<div>
  <select
    name="person"
  >
    <option
      value="Luke Skywalker"
    >
      Luke Skywalker
    </option>
    <option
      value="R2D2"
    >
      R2D2
    </option>
    <option
      value="C-3PO"
    >
      C-3PO
    </option>
    <option
      value="Darth Vader"
    >
      Darth Vader
    </option>
    <option
      value="Yoda"
    >
      Yoda
    </option>
  </select>
</div>
`;
